### PR TITLE
Binheap

### PR DESCRIPTION
**Pull request by jaredscarr**

### CODE REVIEW

File: binheap.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        self.container = []
+        self.current_size = 0
+
+    # def swap(self, value):
+    #     """Swap parent and child."""
+    #     val_index = self.container.index(value)
+    #     current_parent = self.parent(val_index)
+    #     self.container[current_parent] = value
+    #     self.container[val_index] = current_parent
+    #     return self.container[current_parent]
```
Comment (joelstanner) [inline binheap.py:21]: Avoid commented out code in pull requests.

File: test_binheap.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+def test_push():
+    """Test when a value is appended to the heap the structure is maintained."""
+    from binheap import Heap
+    heap = Heap()
+    heap.container = [4, 7, 38, 96]
+    heap.push(5)
+    assert heap.container == [96, 7, 38, 4, 5]
+
```
Comment (joelstanner) [inline test_binheap.py:20]: Make sure to add more tests that cover edge cases. We want to see that your data structure functions correctly at edge cases like popping from an empty heap or a heap with 1 item.

File: binheap.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    def push(self, value):
+        """Push a value to the Heap and maintain Heap structure."""
+        self.container.append(value)
+        child = len(self.container) - 1
+        parent = (child - 1) // 2
+        while child > 1 and self.container[child - 1] > self.container[parent - 1]:
```
Comment (joelstanner) [inline binheap.py:28]: For this class, make sure to adhere to the 80 character line limit. We're mostly on laptops, so screen space is at a premium, and it's part of the linting standards. This is one way to make it work:

``` python
        while (child > 1 and
               self.container[child - 1] > self.container[parent - 1]):
```

See [implicit line joining](https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length)

File: binheap.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+class Heap(object):
+    """Create a Heap."""
+
+    def __init__(self, iterable=None):
+        """Initialize Heap."""
+        # if iterable:
+        #     self.container = list(iterable)
+        #     for i in self.container:
+        #         self.push(i)
+        # else:
```
Comment (joelstanner) [inline binheap.py:11]: Why did you remove this? It's actually part of the assignment description: "The constructor for your heap should default to creating an empty heap, but allow for creating a populated given an iterable as an input."

File: binheap.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            child = parent
+            parent = (child - 1) // 2
+
+    def pop(self):
+        """Pop the root of the heap."""
+        self.container[0] = self.container.pop()
+        self.sort_to_leaf(0)
+
+    def sort_to_leaf(self, index):
+        """Sort up the heap."""
```
Comment (joelstanner) [inline binheap.py:41]: This isn't working correctly. It might help to break down the steps in pseudo-code and place those steps into this function as comments and then write the python for it.

File: test_binheap.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    heap.pop()
+    assert heap.container == [10, 3, 4]
+
+
+def test_push():
+    """Test when a value is appended to the heap the structure is maintained."""
+    from binheap import Heap
+    heap = Heap()
+    heap.container = [4, 7, 38, 96]
+    heap.push(5)
```
Comment (joelstanner) [inline test_binheap.py:18]: Pushing 5 may work, but what about pushing a number larger than 96?
